spock-core/src/main/java/org/spockframework/runtime/DataIteratorFactory.java

```diff
@@ -929,6 +934,26 @@ public List<String> getDataVariableNames() {
       return dataVariableNames;
     }
 
+    /**
+     * Returns how many data providers are processed by this data provider multiplier.
+     *
+     * @return how many data providers are processed by this data provider multiplier
+     */
+    public int getProcessedDataProviders() {
+      int result = 0;
```

Comment:
Should we cache this value in a field, or will this only be called once?

Reply:
Good point thx, also saves the `if`s, as in the constructor it is clear and trivial what to sum up.